src/coreclr/jit/hwintrinsic.h

```diff
@@ -606,6 +606,33 @@ struct HWIntrinsicInfo
         return lookup(id).flags;
     }
 
+    static bool IsXCNTIntrinsic(NamedIntrinsic id)
+    {
+        switch (id)
+        {
+#ifdef TARGET_XARCH
+            case NI_BMI1_TrailingZeroCount:
+            case NI_BMI1_X64_TrailingZeroCount:
+            case NI_LZCNT_LeadingZeroCount:
+            case NI_LZCNT_X64_LeadingZeroCount:
+            case NI_POPCNT_PopCount:
+            case NI_POPCNT_X64_PopCount:
+#elif TARGET_ARM64
+            case NI_AdvSimd_PopCount:
+            case NI_AdvSimd_LeadingZeroCount:
+            case NI_AdvSimd_LeadingSignCount:
+            case NI_ArmBase_LeadingZeroCount:
+            case NI_ArmBase_Arm64_LeadingZeroCount:
+            case NI_ArmBase_Arm64_LeadingSignCount:
+#endif
+                return true;
```

Comment:
Yeah, we can probably do `#if defined(FEATURE_HW_INTRINSICS)` (sp?) around the entire switch

Reply:
Good point

Reply:
> Also, do we need a separate function much for this specific case?

Yeah, I too am of the opinion that it would have probably been better here to inline the code into `IntegralRange::ForNode`, but it is a nit.

Reply:
Updated in f7c53cde166f877b125677b7a425a13c0bbcbd21, d6131e105ff13bd15fb75da0cc03cabca2e913bf